pkg/controller.go

```diff
@@ -102,9 +118,14 @@ func createControllerDeployment(csi *csibaremetalv1.Deployment) *v1.Deployment {
 						}},
 					},
 					Containers:                    createControllerContainers(csi),
+					RestartPolicy:                 corev1.RestartPolicyAlways,
+					DNSPolicy:                     corev1.DNSClusterFirst,
 					TerminationGracePeriodSeconds: pointer.Int64Ptr(TerminationGracePeriodSeconds),
 					NodeSelector:                  csi.Spec.NodeSelectors,
 					ServiceAccountName:            controllerServiceAccountName,
+					DeprecatedServiceAccount:      controllerServiceAccountName,
```

Comment:
Why do we need new fields?

Reply:
This values is filled by client, which we use to create resources. So if we leave it empty or nil, `equality.Semantic.DeepEqual` is not work correctly

Reply:
Actually, there are 2 ways: manually check all fields in `deploymentChanged` or `daemonsetChanged` functions or try to set all default values in recourse creation stage. We have discussed that second is better, so i choose it